pypit/core/artracewave.py

```diff
@@ -0,0 +1,1235 @@
+""" Module for methods related to tracing arc/sky lines across a slit/order
+"""
+from __future__ import (print_function, absolute_import, division, unicode_literals)
+
+import inspect
+import copy
+
+import numpy as np
+
+from scipy import interpolate
+
+import matplotlib.pyplot as plt
+
+from pypit import msgs
+from pypit.core import ararc
+from pypit import arutils
+from pypit import arparse
+from pypit import arpca
+from pypit import arqa
+from pypit import ardebug as debugger
+
+try:
+    from pypit import ginga
+except ImportError:
+    pass
+
+
+def analyze_lines(msarc, trcdict, slit, pixcen, tilt_settings, maskval=-999999.9):
+    # Analyze each spectral line
+    aduse = trcdict["aduse"]
+    arcdet = trcdict["arcdet"]
+    xtfits = trcdict["xtfit"]
+    ytfits = trcdict["ytfit"]
+    wmasks = trcdict["wmask"]
+    badlines = trcdict["badlines"]
+
+    xtilt = np.ones((msarc.shape[1], arcdet.size)) * maskval
+    ytilt = np.ones((msarc.shape[1], arcdet.size)) * maskval
+    mtilt = np.ones((msarc.shape[1], arcdet.size)) * maskval
+    wtilt = np.ones((msarc.shape[1], arcdet.size)) * maskval
+
+    # For displaying later
+    xmodel = []
+    ymodel = []
+
+    for j in range(arcdet.size):
+        if not aduse[j]:
+            continue
+        xtfit = xtfits[j]
+        ytfit = ytfits[j]
+        wmask = wmasks[j]
+        xint = int(xtfit[0])
+        sz = (xtfit.size-1)//2
+
+        # Trim if we are off the detector
+        lastx = min(xint + 2 * sz + 1, msarc.shape[1])
+        if (lastx-xint) < xtfit.size: # Cut down
+            dx = (lastx-xint)-xtfit.size
+            xtfit = xtfit[:dx]
+            ytfit = ytfit[:dx]
+            wmask = wmask[np.where(wmask < (xtfit.size+dx))]
+
+        # Perform a scanning polynomial fit to the tilts
+        wmfit = np.where(ytfit != maskval)
+        if wmfit[0].size > tilt_settings['tilts']['order'] + 1:
+            cmfit = arutils.func_fit(xtfit[wmfit], ytfit[wmfit],
+                                     tilt_settings['tilts']['function'],
+                                     tilt_settings['tilts']['order'],
+                                     minv=0.0, maxv=msarc.shape[1] - 1.0)
+            model = arutils.func_val(cmfit, xtfit, tilt_settings['tilts']['function'],
+                                     minv=0.0, maxv=msarc.shape[1] - 1.0)
+        else:
+            aduse[j] = False
+            badlines += 1
+            continue
+
+        # Can this actually happen??
```

Comment:
I just had a look, and it appears not - so, I think this code can be safely deleted.